cli/src/commands/squash.rs

```diff
@@ -65,12 +79,24 @@ pub(crate) fn cmd_squash(
     args: &SquashArgs,
 ) -> Result<(), CommandError> {
     let mut workspace_command = command.workspace_helper(ui)?;
-    let source = workspace_command.resolve_single_rev(args.revision.as_deref().unwrap_or("@"))?;
-    let mut parents = source.parents();
-    if parents.len() != 1 {
-        return Err(user_error("Cannot squash merge commits"));
+
+    let source;
+    let destination;
+    if args.from.is_some() || args.into.is_some() {
+        source = workspace_command.resolve_single_rev(args.from.as_deref().unwrap_or("@"))?;
+        destination = workspace_command.resolve_single_rev(args.into.as_deref().unwrap_or("@"))?;
+        if source.id() == destination.id() {
+            return Err(user_error("Source and destination cannot be the same."));
```

Comment:
nit: period is usually omitted in error message, but the same message in `cmd_move()` has one.

Reply:
Done.